### PR TITLE
Make Android builds installable on each pull request

### PR DESCRIPTION
**Pull request by tnj**
## Issue
- no issue related

## Overview (Required)
- Upload build artifact (debug dev APK) to DeployGate and add a downloadable link on pull requests
- Once the build is completed, subsequent jobs defined in`Stage*.yml` start running. The jobs have nothing to do with any runnable code in the repository so they can safely use the secret (`DEPLOYGATE_API_KEY`)
- The distribution is automatically deleted once the pull request is closed
- Heavily depend on `gh-scripts` branch in the last year's conference app repo by @jmatsu (might be better to refactor)

## Links
- https://github.com/DroidKaigi/conference-app-2021

## Screenshot
n/a

### CODE REVIEW

**Comment by tnj** [comment]
Due to the GitHub Actions restriction, newly added `on: workflow_run` workflows don't run on the branchs except main. They will start working once they are merged into the main branch.

File: .github/workflows/Build.yml [hunk 1]
```diff
@@ -15,6 +15,14 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  get-pr-number:
+    if: github.event_name == 'pull_request'
+    name: "get-pr-number"
+    runs-on: ubuntu-latest
+    steps:
+      - name: ${{ github.event.pull_request.number }}
+        run: echo ok
```
Comment (takahirom) [inline .github/workflows/Build.yml:24]: Is this debug code? 👀

**Review by takahirom — APPROVED** [review]
LGTM! We can update our app with all your help!

File: .github/workflows/Build.yml (hunk 1 above)
Comment (tnj) [reply .github/workflows/Build.yml:24]: Nope, all we need here is to leave the pull request number for further reference in `Stage*.yml`. Although nothing needs to be run here, steps require doing something.

**Comment by tnj** [comment]
Thanks!

**Pull request merged**